src/main/java/org/eclipse/uprotocol/mqtt/HiveMqMQTT5Client.java

```diff
@@ -221,40 +227,47 @@ public void close() {
     }
 
     private @NotNull MqttTopic getTopicForSending(@NotNull UUri source, @Nullable UUri sink) {
-        return MqttTopic.builder()
+        MqttTopicBuilder.Complete build = MqttTopic.builder()
                 .addLevel(String.valueOf(determinateClientIdentifierFromSource(source)))
 
                 .addLevel(source.getAuthorityName())
                 .addLevel(String.format("%04x", source.getUeId()))
                 .addLevel(String.format("%02x", source.getUeVersionMajor()))
-                .addLevel(String.format("%04x", source.getResourceId()))
-
-                .addLevel(Optional.ofNullable(sink).map(UUri::getAuthorityName).orElse(""))
-                .addLevel(Optional.ofNullable(sink).map(UUri::getUeId).map(ueId -> String.format("%04x", ueId)).orElse(""))
-                .addLevel(Optional.ofNullable(sink).map(UUri::getUeVersionMajor).map(majorVersion -> String.format("%02x", majorVersion)).orElse(""))
-                .addLevel(Optional.ofNullable(sink).map(UUri::getResourceId).map(resourceId -> String.format("%04x", resourceId)).orElse(""))
-
+                .addLevel(String.format("%04x", source.getResourceId()));
```

Comment:
Doesn't this mean that source.resourceId() is the first level?

Reply:
No, it is the 4th level